src/test/scala/mesosphere/marathon/core/launcher/impl/OfferProcessorImplTest.scala

```diff
@@ -7,94 +7,205 @@ import mesosphere.marathon.core.matcher.base.OfferMatcher
 
-    verify(offerMatcher).matchOffer(deadline, offer)
-    verify(taskLauncher).launchTasks(offerId, tasks)
+    Then("we saw the matchOffer request")
+    verify(offerMatcher, times(1)).matchOffer(deadline, offer)
 
+    And("all task launches were rejected")
     assert(dummySource.accepted.isEmpty)
     assert(dummySource.rejected.toSeq.map(_._1) == tasks)
+
+    And("no tasks where stored")
+    noMoreInteractions(taskTracker)
+  }
+
+  test("match successful but first store is so slow that we are hitting storage timeout") {
+    Given("an offer")
+    val dummySource = new DummySource
+    val tasksWithSource = tasks.map(task => TaskWithSource(
+      dummySource, task, MarathonTestHelper.makeTaskFromTaskInfo(task)))
+
+    val offerProcessor = createProcessor()
+
+    val deadline: Timestamp = clock.now() + 1.second
+    And("a cooperative taskLauncher")
+    taskLauncher.launchTasks(offerId, tasks.take(1)) returns true
+
+    And("an cooperative offerMatcher")
+    offerMatcher.matchOffer(deadline, offer) returns Future.successful(MatchedTasks(offerId, tasksWithSource))
+
+    for (task <- tasksWithSource) {
+      taskTracker.store(appId, task.marathonTask) answers { args =>
+        // simulate that stores are really slow
+        clock += 1.hour
+        Future.successful(mock[PersistentEntity])
+      }
+      taskTracker.terminated(appId, task.marathonTask.getId) returns Future.successful(Some(task.marathonTask))
+    }
+
+    When("processing the offer")
+    Await.result(offerProcessor.processOffer(offer), 1.second)
+
+    Then("we saw the matchOffer request and the task launch attempt for the first task")
+    verify(offerMatcher, times(1)).matchOffer(deadline, offer)
+    verify(taskLauncher, times(1)).launchTasks(offerId, tasks.take(1))
+
+    And("one task launch was accepted")
+    assert(dummySource.accepted.toSeq == tasks.take(1))
+
+    And("one task launch was rejected")
+    assert(dummySource.rejected.toSeq.map(_._1) == tasks.drop(1))
+
+    And("the first task was stored")
+    for (task <- tasksWithSource.take(1)) {
+      verify(taskTracker, times(1)).created(appId, task.marathonTask)
+      verify(taskTracker, times(1)).store(appId, task.marathonTask)
+    }
+
+    And("and the second task was not stored")
+    noMoreInteractions(taskTracker)
   }
 
   test("match empty => decline") {
     val offerProcessor = createProcessor()
 
     val deadline: Timestamp = clock.now() + 1.second
-    when(offerMatcher.matchOffer(deadline, offer)).thenReturn(
-      Future.successful(MatchedTasks(offerId, Seq.empty))
-    )
+    offerMatcher.matchOffer(deadline, offer) returns Future.successful(MatchedTasks(offerId, Seq.empty))
 
     Await.result(offerProcessor.processOffer(offer), 1.second)
 
-    verify(offerMatcher).matchOffer(deadline, offer)
-    verify(taskLauncher).declineOffer(offerId, refuseMilliseconds = Some(conf.declineOfferDuration()))
+    verify(offerMatcher, times(1)).matchOffer(deadline, offer)
```

Comment:
It is not added by default in the `Mockito` class that @aquamatthias created and that is used here.